main.go

```diff
@@ -144,6 +148,49 @@ func formatEntry(entry Entry) string {
 	})
 }
 
+// webhookUpdatesPoller starts a HTTP push-delivery method.
+func webhookUpdatesPoller(bot *tgbotapi.BotAPI) (tgbotapi.UpdatesChannel, error) {
+	_, err := bot.SetWebhook(tgbotapi.NewWebhook(config.Webhook))
+	if err != nil {
+		log.Fatalln("Failed to init. webhook:", err)
+	}
+	ch := make(chan tgbotapi.Update, bot.Buffer)
+	router := gin.New()
+	router.Use(gin.Logger())
+	router.POST("/" + bot.Token, func(c *gin.Context) {
+		defer c.Request.Body.Close()
+
+		bytes, err := ioutil.ReadAll(c.Request.Body)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		var update tgbotapi.Update
+		err = json.Unmarshal(bytes, &update)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+
+		ch <- update
+	})
+	go func() {
+		err := router.Run(":" + config.Port)
```

Comment:
Perhaps it is better to have it as a Endpoint option? Say, if I want to run it on 127.0.0.1:8080 and have a reverse proxy forward requests to it.